src/connector/src/source/base.rs

```diff
@@ -21,10 +21,11 @@ use bytes::Bytes;
 use enum_as_inner::EnumAsInner;
 use futures::stream::BoxStream;
 use itertools::Itertools;
+use parking_lot::Mutex;
```

Comment:
Not a real problem but why not `tokio::sync::Mutex`? 👀

Reply:
No issue, but we use `parking_lot` elsewhere. 

Realistically, the mutex is per-actor, so it should not be a performance bottleneck.

I think either is fine.